src/main.ts

```diff
@@ -252,12 +261,25 @@ async function breaking(bufPath: string, inputs: Inputs): Promise<Result> {
 }
 
 // push runs the "buf push" step.
-async function push(bufPath: string, inputs: Inputs): Promise<Result> {
+async function push(
+  bufPath: string,
+  bufVersion: string,
+  inputs: Inputs,
+  moduleNames: string[],
+): Promise<Result> {
   if (!inputs.push) {
     core.info("Skipping push");
     return skip();
   }
+  if (moduleNames.length == 0) {
+    core.info("Skipping push, no named modules detected");
+    return skip();
+  }
   const args = ["push", "--error-format", "github-actions"];
+  // Add --exclude-unnamed on buf versions that support the flag.
+  if (semver.satisfies(bufVersion, ">=1.33.0")) {
```

Comment:
when we 1.0 this action, I would be in favor of just bumping the required minimum version of the whole action to 1.33.0 so we can eliminate this check

Reply:
Yep, any version bumps we can increase the min version on. Left here to allow for a patch on 0.1.